### PR TITLE
ci: add release workflow

### PR DESCRIPTION
**Pull request by kentonshade**
Add a new workflow to release newrelic-php-agent. This workflow uses custom container images for building newrelic-php-agent artifacts for release. 

Note: In this initial PR a private Docker Hub registry is used as source of container images. This is to be replaced with New Relic Docker Hub.

### CODE REVIEW

**Comment by kentonshade** [comment]
@zsistla This most likely will fail.  We'll need to transfer/create secrets, change refs on checkout actions, etc.

**Comment by kentonshade** [comment]
@zsistla So already I see my Docker Hub creds will not be in the upstream secrets.  I prefer not to add those.  We're still waiting on the NR Docker Hub request.

File: .github/workflows/NEW-release-build.yml [hunk 1]
```diff
@@ -0,0 +1,368 @@
+#
+# This is the workflow to do a release build.
+# GHA: 04/07: This is actually replacing the php-release-agent job in jenkins.
+# We do not need the php-release-agent-checkout job.
+# https://issues.newrelic.com/browse/NR-6477
```
Comment (misiekn) [inline .github/workflows/NEW-release-build.yml:5]:
```suggestion
```

**Comment by misiekn** [comment]
Superseded by #623.

**Pull request closed without merging**